### PR TITLE
Add custom datasource for collecting yum/dnf updates

### PR DESCRIPTION
**Pull request by semtexzv**
Dynamically imports os-specific library for handling DNF/YUM interaction.
Performs a resolution of updateable packages, and outputs it as JSON.
Expected performance is on the order of seconds for both RHEL7 and RHEL8.

cc: @MichaelMraka

### CODE REVIEW

File: insights/specs/default.py [hunk 1]
```diff
@@ -964,7 +968,215 @@ def is_mod_loaded_for_ss(broker):
     yum_log = simple_file("/var/log/yum.log")
     yum_repolist = simple_command("/usr/bin/yum -C --noplugins repolist")
     yum_repos_d = glob_file("/etc/yum.repos.d/*.repo")
-    yum_updateinfo = simple_command("/usr/bin/yum -C updateinfo list")
+    #yum_updateinfo = simple_command("/usr/bin/yum -C updateinfo list")
+
+    @datasource(HostContext)
+    def yum_updates(_broker):
+        DEBUG = True
```
Comment (MichaelMraka) [inline insights/specs/default.py:975]: DEBUG = True
should be better switched off for production

Comment (MichaelMraka) [reply insights/specs/default.py:975]: debug  is still on

**Review by MichaelMraka — CHANGES_REQUESTED** [review]

**Review by bfahr — CHANGES_REQUESTED** [review]

**Review by bfahr — CHANGES_REQUESTED** [review]
@semtexzv it looks like you've made all of the requested changes but still need to fix some of your tests.  If you can get those fixed I think it will be ready to merge.

File: insights/parsers/yum_updates.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+from insights.specs import Specs
+
+
+@parser(Specs.yum_updateinfo)
+class YumUpdates(JSONParser):
+    """
+
+    Expected output of the command is::
+
+    {
```
Comment (bfahr) [inline insights/parsers/yum_updates.py:18]: @semtexzv this is causing the unexpected indentation error, it needs to be indented.

**Review by bfahr — CHANGES_REQUESTED** [review]

File: insights/specs/default.py (hunk 1 above)
Comment (semtexzv) [reply insights/specs/default.py]: Removed

File: insights/parsers/yum_updates.py (hunk 2 above)
Comment (semtexzv) [reply insights/parsers/yum_updates.py]: Fixed

**Comment by bfahr** [comment]
@semtexzv this PR will need to be updated after #3088 has been merged.  You can look at that PR to see how the datasources will be moved out of default.py.

**Comment by bfahr** [comment]
I've closed #3088, but added #3110 as a pattern to follow for custom datasources.  Please modify this PR to follow that pattern.

**Comment by semtexzv** [comment]
Thank you, updated. I wasn't sure about testing the datasource, since it invokes local dnf/yum.

**Review by bfahr — APPROVED** [review]

**Review by bfahr — APPROVED** [review]

**Comment by bfahr** [comment]
Verified rebase with master and tests pass.

**Pull request merged**